charts/argo-cd/templates/argocd-server/servicemonitor.yaml

```diff
@@ -12,10 +12,10 @@ metadata:
     {{- with .Values.server.metrics.serviceMonitor.additionalLabels }}
       {{- toYaml . | nindent 4 }}
     {{- end }}
+  {{- with .Values.server.metrics.serviceMonitor.annotations }}
   annotations:
-    {{- range $key, $value := .Values.server.metrics.serviceMonitor.annotations }}
-    {{ $key }}: {{ $value | quote }}
-    {{- end }}
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
```

Comment:
Can we keep the range with key/value please?
I prefere consistency across our YAML. Example:
https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/argocd-applicationset/service.yaml#L5-L10

Reply:
All the other annotations are also like this, example

https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/argocd-repo-server/servicemonitor.yaml#L15-L18

https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/argocd-notifications/servicemonitor.yaml#L15-L18

Reply:
All right, nice find! Then I am okay in implementing it for the ServiceMonitor like your proposal. We can harmonize the codebase afterwards if needed 😄  👍

Reply:
Great! Let me know what else is needed to get this merged. Now I need to do this 😜 

<img width="430" alt="image" src="https://github.com/argoproj/argo-helm/assets/60099717/f05bae75-5361-4df6-88f3-c42835a92d23">